From e7c0904a25fba0876e96c8de697e416a3921898b Mon Sep 17 00:00:00 2001
From: elros34 <elros34@users.noreply.github.com>
Date: Mon, 5 Dec 2022 22:57:46 +0100
Subject: [PATCH] [libconnman-qt] Fix NetworkTechnology objects removal

Before NetworkManager::technologyRemoved is called there is NetworkTechnology::technologyRemoved which destroys interface (delete m_technology) so objPath() returns QString(). This leads to memory leak and flood of dbus GetProperties messages increasing on every technology add/remove.
---
 libconnman-qt/networkmanager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libconnman-qt/networkmanager.cpp b/libconnman-qt/networkmanager.cpp
index 87a6e6f..0bd9622 100644
--- a/libconnman-qt/networkmanager.cpp
+++ b/libconnman-qt/networkmanager.cpp
@@ -900,7 +900,7 @@ void NetworkManager::technologyRemoved(const QDBusObjectPath &technology)
     // if we weren't storing by type() this loop would be unecessary
     // but since this function will be triggered rarely that's fine
     for (NetworkTechnology *net : m_technologiesCache) {
-        if (net->objPath() == technology.path()) {
+        if (net->path() == technology.path()) {
             m_technologiesCache.remove(net->type());
             net->deleteLater();
             break;
